flexget/plugins/filter/torrent_match.py

```diff
@@ -0,0 +1,185 @@
+                log.error('Does this happen?')
+            os.chdir(cwd)
+
+        return entries
+
+    # run first to ensure we have temp files so that torrent plugin can do its magic
+    @plugin.priority(256)
+    def on_task_modify(self, task, config):
+        for entry in task.accepted:
+            if 'torrent' not in entry and 'download' not in task.config:
+                # If the download plugin is not enabled, we need to call it to get
+                # our temp .torrent files
+                download = plugin.get_plugin_by_name('download')
+                download.instance.get_temp_files(task, handle_magnets=True, fail_html=True)
+
+    def prepare_config(self, config):
+        if not isinstance(config['max_size_difference'], float):
+            config['max_size_difference'] = float(config['max_size_difference'].rstrip('%'))
+
+        return config
+
+    # Run first, this is not really an output plugin though, but we need 'torrent' field, which is set in modify
+    @plugin.priority(255)
+    def on_task_output(self, task, config):
+        config = self.prepare_config(config)
+        max_size_difference = config['max_size_difference']
+        local_entries = self.get_local_files(config, task)
+
+        matched_entries = set()
+        for entry in task.accepted:
+            if 'torrent' not in entry:
+                log.debug('Skipping entry %s as it is not a torrent file', entry['title'])
+                continue
+
+            torrent_files = []
+            for item in entry['torrent'].get_filelist():
+                # if torrent is a multi_file, prepend the name
+                path = os.path.join(item['path'], item['name'])
+                if entry['torrent'].is_multi_file:
+                    path = os.path.join(entry['torrent'].name, path)
+
+                torrent_files.append(TorrentMatchFile(path, item['size']))
+
+            for local_entry in local_entries:
+                local_files = local_entry['files']
+
+                # skip root dir of the local entry if torrent is single file or if name_in_path is false
+                skip_root_dir = not entry['torrent'].is_multi_file and local_entry['files_root'] or \
+                    not config['name_in_path']
+
+                matches = 0
+                missing_size = 0
+                total_size = 0
+                for f in torrent_files:
+                    for local_file in local_files:
+                        # remove the root dir from the local path
+                        local_path = local_file.path
+                        if skip_root_dir:
+                            local_path = os.path.relpath(local_path, local_entry['files_root'])
+                        # if f == local_file, break out of inner loop and increment match counter
+                        # we force sizes to be exact, TODO is this correct?
+                        # TODO allow root dir to differ? Requires torrent clients have "do not add name to path"
```

Comment:
I don't think any of our client plugins support the "do not add name to path" and to keep things simple I'd prefer not supporting it anyway.